web/src/features/App/api/useLicense.ts

```diff
@@ -7,24 +7,20 @@ function useLicenseWithIntercept() {
 
   let timerId = useRef<null | NodeJS.Timeout>(null);
 
-  // DEV: sleep function to delay the return of the data
-  // use this to simulate a slow loading request
-  // const sleep = (ms = 1000): Promise<void> => {
-  //   return new Promise<void>((resolve) => {
-  //     setTimeout(() => resolve(), ms);
-  //   });
-  // };
-
   let licenseQuery = useLicense();
 
   useEffect(() => {
     axios.interceptors.request.use(
       (x) => {
-        // not sure if this check is enough to make sure that it
-        // only happens on /license endpoint
         if (x.url?.endsWith("/license")) {
+          if (timerId.current) {
+            clearTimeout(timerId.current);
+            timerId.current = null;
+          }
```

Comment:
Needed to add timeout cleanups to this for handling subsequent requests. Without this we'd get an orphaned `setTimeout` would continue to run and set the slowLoading state.